zkevm-circuits/src/evm_circuit/util/common_gadget.rs

```diff
@@ -309,7 +309,7 @@ impl<F: Field, const N_ADDENDS: usize, const INCREASE: bool>
     pub(crate) fn construct(
         cb: &mut EVMConstraintBuilder<F>,
         address: WordLoHi<Expression<F>>,
-        updates: Vec<Word32Cell<F>>,
+        updates: &[Word32Cell<F>],
```

Comment:
Just curious, what is the purpose of using `&[]` instead of `Vec`?

Reply:
`Vec` takes a vector for argument.
`[]` takes a vector or slice for argument. If we already know the number of elements in an array, we can use slice `[]` instead of a vector `vec![]`. The intention here was to make lines shorter but not memory optimization.